src/vs/editor/contrib/inlineCompletions/browser/inlineCompletionsModel.ts

```diff
@@ -486,6 +469,34 @@ export class InlineCompletionsModel extends Disposable {
 	}
 }
 
+export function getSecondaryEdits(textModel: ITextModel, positions: readonly Position[], primaryEdit: SingleTextEdit): SingleTextEdit[] {
+	const primaryPosition = positions[0];
+	const secondaryPositions = positions.slice(1);
+	const primaryEditEndPosition = primaryEdit.range.getEndPosition();
+	const replacedTextAfterPrimaryCursor = textModel.getValueInRange(
+		Range.fromPositions(primaryPosition, primaryEditEndPosition)
+	);
+	const positionWithinTextEdit = subtractPositions(primaryPosition, primaryEdit.range.getStartPosition());
+	const secondaryEditText = getTextFromPosition(primaryEdit.text, positionWithinTextEdit);
+	return secondaryPositions.map(pos => {
+		const textAfterSecondaryCursor = textModel.getValueInRange(
+			Range.fromPositions(pos, primaryEditEndPosition)
+		);
+		const l = commonPrefixLength(replacedTextAfterPrimaryCursor, textAfterSecondaryCursor);
+		const range = Range.fromPositions(pos, pos.delta(0, l));
+		return new SingleTextEdit(range, secondaryEditText);
+	});
+}
+
+function getTextFromPosition(text: string, pos: Position): string {
```

Comment:
I think `substringPos` would be a good name for this function.

Reply:
Also, I think this function could be more performant - you don't have to split the actual lines.
Just convert `pos` into an offset and do one single substring call on `text`.